code/game/objects/items.dm

```diff
@@ -85,8 +85,8 @@ var/global/image/fire_overlay = image("icon" = 'icons/goonstation/effects/fire.d
 	if(ismob(loc))
 		var/mob/m = loc
 		m.unEquip(src, 1)
-	for(var/X in actions)
-		qdel(X)
+	QDEL_LIST(actions)
+	master = null

```

Comment:
To further explain, basically this is a way of hooking an assembly or assembly holder to a master item, then abusing it for updating its icon or (and this only goes for tanks), `qdel`ing them.

I understand why it exists, but it seems this should be on the individual objects that use it (or not at all), than have a weird holder like this.